### PR TITLE
Device

### PR DESCRIPTION
**Pull request by skinnyiam**
check this , device-id for all the simple-analytics api's @rosekamallove

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "@trpc/next": "^10.35.0",
     "@trpc/react-query": "^10.35.0",
     "@trpc/server": "^10.35.0",
+    "@types/device-uuid": "^1.0.0",
```
Comment (rosekamallove) [inline package.json:41]: Remove this package.

File: src/components/CoursePublicPage.tsx [hunk 2]
```diff
@@ -66,17 +66,40 @@ const CoursePublicPage = ({ course, creator, preview }: Props) => {
 
   const currentUrl = router.asPath;
 
+  function generateDeviceId() {
```
Comment (rosekamallove) [inline src/components/CoursePublicPage.tsx:69]: helper wale folder mei file banake usme dal ise

File: src/components/CoursePublicPage.tsx [hunk 3]
```diff
@@ -66,17 +66,40 @@ const CoursePublicPage = ({ course, creator, preview }: Props) => {
...
+    const timestamp = Date.now().toString(36);
+    const randomString = Math.random().toString(36).substring(2, 10);
+    return `${timestamp}-${randomString}`;
+  }
+  function getDeviceId() {
```
Comment (rosekamallove) [inline src/components/CoursePublicPage.tsx:74]: Yeh wala bhi

File: src/pages/[creator_id]/course/[id]/index.tsx [hunk 4]
```diff
@@ -40,37 +40,59 @@ const Index = ({ courseId, creatorProfile }: Props) => {
...
-    useState(true);
+  // const [isFirstTimeCoursePagedViewed, setIsFirstTimeCoursePagedViewed] =
+  //   useState(true);
 
   const { data: course } = api.course.getCourse.useQuery({ id: courseId });
 
   const currentUrl = router.asPath;
   const userId = session.data?.user.id;
 
+  function generateDeviceId() {
```
Comment (rosekamallove) [inline src/pages/[creator_id]/course/[id]/index.tsx:51]: Do jgh kese aa gya yeh function?

Comment (rosekamallove) [reply src/pages/[creator_id]/course/[id]/index.tsx:51]: Idhar localstorage bhi ni dikhra mereko.

**Review by rosekamallove — CHANGES_REQUESTED** [review]

**Comment by rosekamallove** [comment]
Count krke bhi show krna hai backend se course analytics mei.

File: src/pages/[creator_id]/course/[id]/index.tsx (hunk 4 above)
Comment (skinnyiam) [reply src/pages/[creator_id]/course/[id]/index.tsx:51]: sab saaf kar diya hai

**Pull request merged**